app/src/main/java/com/android/solvit/seeker/ui/review/CreateReview.kt

```diff
@@ -0,0 +1,298 @@
+  val context = LocalContext.current
+  DisposableEffect(Unit) {
+    val activity = context as? ComponentActivity
+    activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+    onDispose { activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED }
+  }
+
+  Scaffold(topBar = { TopSection(navigationActions) }) { paddingValues ->
+    Column(
+        modifier =
+            Modifier.padding(paddingValues)
+                .fillMaxWidth()
+                .padding(16.dp)
+                .verticalScroll(rememberScrollState()),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally) {
+          RequestBox(requestState, listProviderViewModel, navigationActions)
+          RatingBar(ratingState)
+          OutlinedTextField(
+              value = comment,
+              onValueChange = { comment = it },
+              modifier = Modifier.fillMaxWidth().height(160.dp).testTag("reviewComment"),
+              label = { Text("Comment") },
+              placeholder = { Text("Leave a comment") },
+              shape = RoundedCornerShape(16.dp),
+              colors =
+                  OutlinedTextFieldDefaults.colors(
+                      unfocusedContainerColor = Color.Transparent,
+                      focusedBorderColor = colorScheme.secondary,
+                      unfocusedBorderColor = colorScheme.onSurfaceVariant))
+          Spacer(modifier = Modifier.size(16.dp))
+          Button(
+              onClick = {
+                if (requestState.value != null && requestState.value!!.providerId != null) {
+                  val review =
+                      Review(
+                          uid = reviewViewModel.getNewUid(),
+                          authorId = requestState.value!!.userId,
+                          serviceRequestId = requestState.value!!.uid,
+                          providerId = requestState.value!!.providerId!!,
+                          rating = ratingState.intValue,
+                          comment = comment)
+                  reviewViewModel.addReview(review)
+                  Toast.makeText(context, "Review Submitted", Toast.LENGTH_SHORT).show()
+                } else {
+                  Toast.makeText(context, "Error Submitting Review", Toast.LENGTH_SHORT).show()
+                }
+                navigationActions.goBack()
+              },
+              modifier = Modifier.testTag("submitReviewButton")) {
+                Row(
+                    horizontalArrangement = Arrangement.spacedBy(16.dp),
+                    verticalAlignment = Alignment.CenterVertically) {
+                      Text("Submit Your Review")
+                      Icon(imageVector = Icons.Filled.Done, contentDescription = "Submit Review")
+                    }
+              }
+        }
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TopSection(navigationActions: NavigationActions) {
+  TopAppBar(
```

Comment:
This method can be removed and replaced with the `GoBackButton` method, which performs the same function but also handles the issue of double-clicking the button.